de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/ad/window/api/impl/ADWindowDAO.java

```diff
@@ -394,12 +395,14 @@ public void copyWindow(@NonNull final AdWindowId targetWindowId, @NonNull final
 		logger.debug("Copying from: {} to: {}", sourceWindow, targetWindow);
 
 		copy()
-				.setSkipCalculatedColumns(true)
-				.addTargetColumnNameToSkip(I_AD_Window.COLUMNNAME_Name)
-				.addTargetColumnNameToSkip(I_AD_Window.COLUMNNAME_InternalName)
-				.setFrom(sourceWindow)
-				.setTo(targetWindow)
-				.copy();
+		.setSkipCalculatedColumns(true)
+		// skip it because other wise the MV will fill the name from the AD_Element of the original window and unique name constraint will be broken
+		.addTargetColumnNameToSkip(I_AD_Window.COLUMNNAME_AD_Element_ID)
```

Comment:
so how the window name / element will be generated then?

Reply:
@teosarca  AD_Element_ID is mandatory and you need to set when you create AD_window.
When you run the process, if ad_element_id is not skipped, it will be set from the original window and we will get 2 windows with the same name. But this is forbidden because it will have a UC for windows name, si everything will be rolled back.
So, the process will not work at all!